### PR TITLE
deploy to GH when tag a release

### PR DESCRIPTION
**Pull request by cpanato**
Adding a functionality to deploy to GH when we tag a release

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,28 +3,78 @@ jobs:
...
     steps:
       - checkout
-
       - run: curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
       - run: make test
 
+  build:
+    docker:
+      - image: circleci/golang:1.12-node
+    working_directory: /go/src/github.com/mattermost/mattermost-plugin-jira
```
Comment (hanzei) [inline .circleci/config.yml]: See my comment in https://github.com/mattermost/mattermost-plugin-github/pull/60#discussion_r264269279.

Otherwise LGTM.

Comment (cpanato) [reply .circleci/config.yml]: done @hanzei

**Pull request merged**